src/Angor/Avalonia/Angor.Contexts.Funding/Investor/Operations/GetInvestorProjectRecovery.cs

```diff
@@ -16,84 +18,315 @@ public static class GetInvestorProjectRecovery
-                ScriptType = ProjectScriptTypeEnum.InvestorWithPenalty,
-            });
-
-            // 4) Spent by founder
-            dto.Items.Add(new InvestorStageItemDto
+                var recoveryTansaction =
+                    await indexerService.GetTransactionInfoByIdAsync(lookup.Value.RecoveryTransactionId);
+                if (recoveryTansaction != null)
+                    penaltyExpieryDate = Utils.UnixTimeToDateTime(recoveryTansaction.Timestamp)
+                        .AddDays(projectInfo.PenaltyDays);
+            }
+
+
+            var tasks = stageItems.Select(item => CheckTransactionSpendingAsync(transactionInfo, item, lookup, penaltyExpieryDate,
+                investmentTransaction, projectInfo));
+            
+            await Task.WhenAll(tasks);
+
+            var response = new InvestorProjectRecoveryDto
             {
-                StageIndex = 3,
-                Amount = 250_0000,
-                IsSpent = true,
-                Status = "Spent by founder",
-                ScriptType = ProjectScriptTypeEnum.Founder,
-            });
-
-            // 5) Spent by investor (unfunded release)
-            dto.Items.Add(new InvestorStageItemDto
+                CanRecover = stageItems.Any(a => a.IsSpent == false),
+                CanRelease = (stageItems.Any(a => a.ScriptType == ProjectScriptTypeEnum.InvestorWithPenalty) &&
+                              DateTime.UtcNow > penaltyExpieryDate),
+                TotalSpendable = stageItems.Where(a => !a.IsSpent).Sum(a => a.Amount),
+                TotalInPenalty = stageItems.Where(t => t.ScriptType == ProjectScriptTypeEnum.InvestorWithPenalty)
+                    .Sum(t => t.Amount),
+                EndOfProject = projectInfo.ExpiryDate < DateTime.Now,
+                ExpiryDate = projectInfo.ExpiryDate,
+                Name = project.Name,
+                PenaltyDays = projectInfo.PenaltyDays,
+                ProjectIdentifier = project.Id.Value,
+                Items = stageItems.ToList()
+            };
+
+            return Result.Success(response);
+        }
+
+        private async Task<Result> CheckTransactionSpendingAsync(QueryTransaction transactionInfo, InvestorStageItemDto item, Result<InvestmentSpendingLookup> lookup,
+            DateTimeOffset penaltyExpieryDate, Transaction? investmentTransaction, ProjectInfo projectInfo)
+        {
+            // if (unconfirmedOutbound.ContainsOutpoint(new Outpoint(StageInfo.TransactionInfo.TransactionId, item.Outputindex)))
+            // {
+            //     item.IsSpent = true;
+            //     item.SpentTo = "pending confirmations";
+            //     item.ProjectScriptType = new ProjectScriptType { ScriptType = ProjectScriptTypeEnum.Unknown };
+            //
+            //     continue;
+            // }
+            
+            var output = transactionInfo.Outputs.ElementAt(item.StageIndex + 2);
+
+            if (!string.IsNullOrEmpty(output.SpentInTransaction))
```

Comment:
this code looks correct to me, I assume it is a copy paste from the web page right?

Reply:
Yes copied and pasted with adjustments to an API flow